tests/test_api.py

```diff
@@ -27,145 +24,67 @@ def bliss_client(cwd, cfg):
+    def test_load_pretrained_weights(self, bliss_client, monkeypatch):
+        monkeypatch.setattr(requests, "get", mock_get)
+        monkeypatch.setattr(requests, "post", mock_post)
 
-    base_url = f"https://api.github.com/repos/prob-ml/bliss/contents/data/sdss/{run}/{camcol}"
-    save_path_rc = Path(cwd) / f"data/sdss/{run}/{camcol}"
-    save_path_rcf = save_path_rc / f"{field}"
-    save_path_rcf.mkdir(parents=True, exist_ok=True)
-
-    assert os.environ["GITHUB_TOKEN"] is not None, "GITHUB_TOKEN environment variable not found"
-    headers = {"Authorization": f"Bearer {os.environ['GITHUB_TOKEN']}"}
-    pf = download_git_lfs_file(f"{base_url}/photoField-{run6}-{camcol}.fits", headers)
-    po = download_git_lfs_file(
-        f"{base_url}/{field}/photoObj-{run6}-{camcol}-{field4}.fits", headers
-    )
-    psf = download_git_lfs_file(
-        f"{base_url}/{field}/psField-{run6}-{camcol}-{field4}.fits", headers
-    )
-    save_file(f"{save_path_rc}/photoField-{run6}-{camcol}.fits", pf)
-    save_file(f"{save_path_rc}/{field}/photoObj-{run6}-{camcol}-{field4}.fits", po)
-    save_file(f"{save_path_rc}/{field}/psField-{run6}-{camcol}-{field4}.fits", psf)
-
-    for band in "ugriz":
-        fpm = download_git_lfs_file(
-            f"{base_url}/{field}/fpM-{run6}-{band}{camcol}-{field4}.fits", headers
+        filename = "sdss_pretrained.pt"
+        bliss_client.load_pretrained_weights_for_survey(
+            survey="sdss",
+            filename=filename,
+            request_headers={"Authorization": "test"},
         )
-        frame = download_git_lfs_file(
-            f"{base_url}/{field}/frame-{band}-{run6}-{camcol}-{field4}.fits", headers
+        not_found_err_msg = (
+            "pretrained weights "
+            + f"{bliss_client.cwd}/data/pretrained_models/{filename} "
+            + "not found"
         )
-        save_file(f"{save_path_rcf}/fpM-{run6}-{band}{camcol}-{field4}.fits", fpm)
-        save_file(f"{save_path_rcf}/frame-{band}-{run6}-{camcol}-{field4}.fits", frame)
-
-
-def save_file(file_loc, data):
-    with open(file_loc, "wb") as f:
-        f.write(data)
+        assert Path(
+            bliss_client.cwd + f"/data/pretrained_models/{filename}"
+        ).exists(), not_found_err_msg
+
+    def test_train(self, bliss_client, monkeypatch):
+        monkeypatch.setattr(api, "_train", mock_train)
+        bliss_client.train("test", training={"n_epochs": 1})
+
+    def test_train_on_cached_data(self, cfg, bliss_client, monkeypatch):
+        monkeypatch.setattr(api, "_train", mock_train)
+        bliss_client.train_on_cached_data("test.pt", 4, 4, None, None, "test_pretrained.pt")
+
+    def test_load_survey(self, bliss_client, monkeypatch):
+        monkeypatch.setattr(requests, "get", mock_get)
+        monkeypatch.setattr(requests, "post", mock_post)
+        monkeypatch.setattr(gzip, "decompress", lambda x: x)
+        monkeypatch.setattr(bz2, "decompress", lambda x: x)
+
+        download_dir = "data"
```

Comment:
Should this be "data/sdss" instead?

Reply:
Doesn't matter what we call it, it's just a temporary directory we download to

Reply:
I think we prefer “data/sdss” to stay consistent with whatever our repo is set up with

Reply:
Yup; nit: but I think we should prefer “data/sdss” to stay consistent with whatever our repo is set up with? But probably could just be done in a later PR that expands this test’s functionality—e.g., when API changes due to generalizing the predict function.